File: es.kubenet.K8sManager.Infraestructure/Dto/Employees/EmployeeDTO.cs
using es.kubenet.K8sManager.Infraestructure.Dto.Departments;
using System;
using System.ComponentModel.DataAnnotations;

namespace es.kubenet.K8sManager.Infraestructure.Dto.Employees
{
  public class EmployeeDTO
  {
    public Guid Id { get; set; }

    [Required]
    public string Name { get; set; } = null!;
    public string? Surname1 { get; set; }
    public string? Surname2 { get; set; }

    [Required]
    public string UserName { get; set; } = null!;


    [Required]
    public DepartmentDTO Department { get; set; } = null!;
    [Required]
    public Guid DepartmentId { get; set; }
  }
}

File: es.kubenet.K8sManager.Angular/src/app/shared/api/models/employee-dto.ts
/* tslint:disable */
import { DepartmentDto } from './department-dto';
export interface EmployeeDto {
  department: DepartmentDto;
  departmentId: string;
  id?: string;
  name: string;
  surname1?: null | string;
  surname2?: null | string;
  userName: string;
}
